sql/table.cc

```diff
@@ -450,6 +450,19 @@ void TABLE_SHARE::destroy()
 
   DBUG_ENTER("TABLE_SHARE::destroy");
   DBUG_PRINT("info", ("db: %s table: %s", db.str, table_name.str));
+  Field* current_field;
+  for (uint i = 0; i < fields; ++i)
+  {
+    current_field = field[i];
+    if (current_field->has_associated_compression_dictionary())
+    {
+      my_free(const_cast<char*>(current_field->zip_dict_data.str));
```

Comment:
hmm maybe LEX_CSTRING is not the best choice here, then. Not sure, please re-check and decide